server/src/main/java/com/google/musicanalysis/util/Secrets.java

```diff
@@ -0,0 +1,36 @@
+package com.google.musicanalysis.util;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretVersionName;
+
+public final class Secrets {
+  private static Object SYNC_OBJECT = new Object();
+  private static SecretManagerServiceClient CLIENT;
+
+  private static SecretManagerServiceClient getClient() throws IOException {
+    var localRef = CLIENT;
```

Comment:
nit: this localRef variable is unnecessary. You can just use CLIENT directly.

Reply:
Isn't it possible for the value of `CLIENT` to change between the time we check whether it is null and the time we enter the `synchronized` block?

Reply:
Sure, but you're already checking if it's null a second time on lines 17 and 18